gph/python/ripser_interface.py

```diff
@@ -106,6 +107,21 @@ def _collapse_coo(row, col, data, thresh):
             np.hstack([data_diag, data]))
 
 
+def _collapse_dense(dm, thresh):
+    """Run edge collapser on off-diagonal data and then reinsert diagonal
+    data if any non zero values is present"""
+    row, col, data = gph_collapser.flag_complex_collapse_edges_dense(
+        dm.astype(np.float32), thresh)
```

Comment:
I think we should downcast `dm` before passing it to `_collapse_dense`.  Currently, the final `data` might end up being 64-bit.

Reply:
I am not sure I get what you mean.
You want to downcast just before calling `_collapse_dense`, i.e. `_collapse_dense(dm.astype(np.float32), thresh)` ?
Because as far as I understand, we do the same thing in `_collapse_coo` to downcast just before calling collapser.

IMO it is better to downcast inside of `_collapse_dense`, it keeps behavior encapsulated. And, at some point we will support both 32 and 64-bit precision for values in the distance matrix.

Reply:
I was thinking of calling `_collapse_dense(dm.astype(np.float32), thresh)`, but I would also be happy with 
```
def _collapse_dense(dm, thresh):
    """Run edge collapser on off-diagonal data and then reinsert diagonal
    data if any non-zero value is present."""
    dm = dm.astype(np.float32)
    row, col, data = \
        gph_collapser.flag_complex_collapse_edges_dense(dm, thresh)
```
The point is that I don't like the fact that in line 116 `data_diag` could be 64-bit.

Reply:
Nice spotted !
You are right, I think that doing it inside of the function is better.
Because we hide some internals that I hope will be updated this year !
If you don't mind, I will proceed with my suggestion, I am happy if you can review it ;).

Reply:
Changed pushed.
Repeated becnhmark, no regression from my latest one.

Reply:
Perfect!